sql: support pg_column_size

C0rWin opened the pull request:
This commit adds `pg_column_size` builtin command to measure number of
bytes allocated by datum.

Signed-off-by: Artem Barger <bartem@il.ibm.com>

Release note (sql change): enable pg_column_size builtin command to
count amount of bytes stored by column

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55312)
<!-- Reviewable:end -->

C0rWin commented:
An attempt to address #23173

jordanlewis reviewed (commented):
Thanks @C0rWin. I'm realizing that I think the Postgres definition for this builtin is actually about how much space is used on-disk for a column, and what I suggested is about in-memory space.

I think if we used `EncodeTableValue`, we'd have something closer to what Postgres does.

```
	 len(EncodeTableValue(nil, descpb.ColumnID(encoding.NoColumnID), inputDatum, nil))
```

C0rWin commented:
@jordanlewis indeed, I've realized that while comparing results with Postgres, I will give it a try based on your suggestion. Thanks for reviewing PR.

C0rWin commented:
@jordanlewis I've updated PR based on your suggestion, hope it's fine now.

C0rWin committed:
sql: support pg_column_size

This commit adds `pg_column_size` builtin command to measure number of
bytes allocated by datum.

Signed-off-by: Artem Barger <bartem@il.ibm.com>

Release note (sql change): enable pg_column_size builtin command to
count amount of bytes stored by column

jordanlewis reviewed (approved):
LGTM, thanks!

bors r+

C0rWin commented:
> === RUN   TestSchemaChangeFailureAfterCheckpointing
> ThreadSanitizer: clock allocator overflow (65536*1024). Dying.
> FAIL	github.com/cockroachdb/cockroach/pkg/sql	1348.974s

CI failure doesn't seem related to my PR :/
